Cache clear can now rely on cache tag invalidation

Adnan-cds opened the pull request:
This comment in the dropped code says it best:
> Better to use cache tags instead of doing a full flush?"

#resolves #126

Adnan-cds committed:
Cache clear can now rely on cache tag invalidation.

This comment in the dropped code says it best:
"Better to use cache tags instead of doing a full flush?" :)

andybroomfield commented:
> 1) Drupal\Tests\localgov_alert_banner\Functional\AlertConfirmationTest::testAlertConfirmation
The link Remove banner was not found on the page.

Failing consistently so not a Github tests issue. This implies the confirmation form for removing a banner is not cache invalidating properley.

Adnan-cds committed:
Revert "Cache clear can now rely on cache tag invalidation."

This reverts commit 430d0e6b13426c16dc965ba1a9098a29459bf00b.

Adnan-cds committed:
Cache clear improvement.

Updating an Alert banner now resorts to less destructive route rebuilding
instead of rebuilding the entire cache.

Adnan-cds commented:
Hi Andy,
The "Put banner live/Remove banner" route is using an [outdated copy of the Alert banner entity](https://github.com/localgovdrupal/localgov_alert_banner/blob/a0f2c7dfee80c8f393ac3327d86fce8f85571a79/src/Controller/AlertBannerEntityController.php#L221) during the test run.  I couldn't pin point the exact cause.  I have replaced drupal_flush_all_caches() with the less destructive [router rebuild](https://git.drupalcode.org/project/drupal/-/blob/9.2.x/core/lib/Drupal/Core/Routing/RouteBuilderInterface.php#L43).

andybroomfield reviewed (approved):
Ok, this works for me.
